globus_cli/services/transfer/async_delete.py

```diff
@@ -18,26 +18,24 @@
                      'asynchronous task.'))
 @common_options
 @submission_id_option
-@endpoint_id_option(help='ID of the Endpoint from which to delete file(s)')
-@click.option('--path', help='Path to the file/dir to delete')
 @click.option('--recursive', is_flag=True, help='Recursively delete dirs')
 @click.option('--ignore-missing', is_flag=True,
               help="Don't throw errors if the file or dir is absent")
 @click.option('--batch', is_flag=True,
               help=('Accept a batch of paths on stdin (i.e. run in '
-                    'batchmode). Uses --endpoint-id as passed on the '
-                    'commandline.'))
-def async_delete_command(batch, ignore_missing, recursive, path, endpoint_id,
-                         submission_id):
+                    'batchmode). Uses ENDPOINT_ID as passed on the '
+                    'commandline, and includes any commandline PATH given'))
+@click.argument('endpoint_plus_path', metavar=ENDPOINT_PLUS_OPTPATH.metavar,
+                type=ENDPOINT_PLUS_OPTPATH)
+def async_delete_command(batch, ignore_missing, recursive, submission_id,
+                         endpoint_plus_path):
     """
     Executor for `globus transfer async-delete`
     """
+    endpoint_id, path = endpoint_plus_path
     if path is None and (not batch):
```

Comment:
Ah, I had incorrectly remembered the behavior you described (and which, at the time, we agreed upon) for batchmode with paths.

I think doing that correctly will be somewhat delicate, to respect absolute paths and produce normalized paths. Can we leave the behavior I put in for now and fix it before doing a release?
Or make it more strict now and change it later?
I think there are all kinds of interesting questions we'll be asking ourselves, like what to do when someone does a batch delete with `path="../../"`.